plugin/save_command.py

```diff
@@ -55,11 +55,12 @@ def _on_complete(self) -> None:
             self._on_done()
 
     def _purge_changes_if_needed(self) -> None:
+        # TODO: Does this run in the async thread?
```

Comment:
Yes and no.

It generally runs on the main thread but for the `WillSaveWaitTask` it also happens to be triggered from the custom async thread.

Should be no problem to straighten it up though.

Reply:
Well, it might be a bit of work as whole run-task-on-save should be switched run async...

Reply:
Pushed 8bf0840 which should address that. Save tasks should now run on ST's async thread and so the `purge_changes` call should be in sync now.